src/experiments/misc/Raja_experiment.ml

```diff
@@ -43,7 +44,9 @@ let ranges_of_path (path : Fpath.t) : Function_range.ranges =
          * the rule_id of the match. That would require to pass
          * more info to ranges_of_path() though.
          *)
-        try Parse_target.parse_program (Fpath.to_string path) with
+        try Parse_target.parse_program !!path with
+        (* TODO: does this still catch what it's supposed to catch
+           since we no longer use List.hd? *)
         | Failure "hd" -> []
```

Comment:
@aryx I need your help here.

Reply:
yes, I'll fix your branch today hopefully with the right fix. We should throw an exn like
CantInferLangFromFilename in Parse_target.parse_program instead of abusing Failure "hd".